language/move-vm/runtime/src/loader.rs

```diff
@@ -580,6 +580,14 @@ impl Loader {
             .collect::<PartialVMResult<Vec<_>>>()
             .map_err(|err| err.finish(Location::Undefined))?;
 
+        let return_tys = func
+            .return_
+            .0
+            .iter()
+            .map(|tok| self.module_cache.lock().make_type(module.module(), tok))
```

Comment:
This acquires a lock in a loop if there are multiple return values... Is it possible to acquire it once instead?

Reply:
Good catch. Will also get the one above (code to make parameter types) fixed.

Reply:
This ended up not working due to deadlock and I had to revert this back. We should however take a more systematic look at the loader and see if we can further simplify it a bit -- potentially changing the way concurrency is modeled.